tests/testing_common.py

```diff
@@ -430,6 +430,26 @@ def _test_from_pretrained_config_construction(self, model_id, config_cls, config
             self.assertTrue(model_from_pretrained.peft_config["default"].inference_mode)
             self.assertIs(model_from_pretrained.peft_config["default"], config)
 
+    def _test_merge_layers_fp16(self, model_id, config_cls, config_kwargs):
```

Comment:
Small issue with this test: When it is run locally on a cuda-enabled device, it will automatically use the GPU and not CPU, right? Therefore, it wouldn't really test the casting back and forth. On CI, it will run correctly though. Not sure if it's worth fixing.

Reply:
Yes great point, I have forced the test to happen on CPU